api/scpca_portal/models/project.py

```diff
@@ -1159,3 +1159,13 @@ def update_downloadable_sample_count(self):
         self.downloadable_sample_count = downloadable_sample_count
         self.unavailable_samples_count = unavailable_samples_count
         self.save()
+
+    def get_filtered_data_file_paths(self, download_config: Dict) -> List[Path]:
+        if download_config["includes_merged"]:
+            return [Path(fp) for fp in self.data_file_paths]
+
+        return [
+            file_path
+            for file_path in [Path(fp) for fp in self.data_file_paths]
```

Comment:
It might make sense to rename `data_file_paths` as `data_files` and create a property `data_file_paths` which handles the converting the array of strings to an array of lib paths.

Reply:
What about adding this to the `Library` and `Project` models? This accomplishes the above and allows us to work exclusively with `Path` objects.

```python
def __getattribute__(self, name):
    if name == "data_file_paths":
        raw_paths = super().__getattribute__(name)
        return [Path(p) for p in raw_paths]
    return super().__getattribute__(name)
```

Reply:
Let's write an issue abstracting the `data_file_path` attribute and its assorted logic into a base class (per our conversation). Also, let's look into a custom model field to handle what's being discussed here (converting all entries from a str to a Path object).